cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1223,35 +1223,28 @@ private KafkaVersionChange getKafkaVersionChange(StatefulSet kafkaSts) {
         Future<ReconciliationState> getZookeeperDescription() {
             Promise<ReconciliationState> promise = Promise.promise();
 
-            vertx.createSharedWorkerExecutor("kubernetes-ops-pool").executeBlocking(
-                future -> {
-                    try {
-                        StatefulSet sts = zkSetOperations.get(namespace, ZookeeperCluster.zookeeperClusterName(name));
-                        Storage oldStorage = getOldStorage(sts);
+            Future<StatefulSet> futSts = zkSetOperations.getAsync(namespace, ZookeeperCluster.zookeeperClusterName(name));
```

Comment:
I think we can avoid the local variable here, just chaining the `setHandler` on the `getAsync` call.